pyccel/codegen/printing/cwrappercode.py

```diff
@@ -1090,31 +1102,62 @@ def _print_Interface(self, expr):
+            A dictionary linking the parsed arguments to the possible types.
+            The values are tuples containing the expected argument, the type check, and the flag.
+
+        func_name : str
+            The name of the interface.
+
+        Returns
+        -------
+        FunctionDef
+            The function which determines the Interface key.
+        """
         check_func_body = []
-        flags = (len(types_dict) - 1) * 4
         for arg in types_dict:
             var_name = ""
             body = []
-            types = []
             arg_type_check_list = list(types_dict[arg])
-            arg_type_check_list.sort(key= lambda x : x[0].precision)
-            for elem in arg_type_check_list:
+            types = [elem[0] for elem in arg_type_check_list]
+            precisions = [f'{v.precision * 8} bit' if v.precision != -1 else 'native' for v in types]
+            type_strs = [str_dtype(v.dtype) for v in types]
+            error = ' or '.join([f'{p} {t}' if not isinstance(v.dtype, NativeBool) else t
+                        for p,t,v in zip(precisions, type_strs, types)])
+            if len(arg_type_check_list) == 1:
+                elem = arg_type_check_list[0]
                 var_name = elem[0].name
-                value = elem[2] << flags
-                body.append(IfSection(elem[1], [AugAssign(check_var, '+' ,value)]))
-                types.append(elem[0])
-            flags -= 4
-            error = ' or '.join(['{} {}'.format(str(v.precision * 8)+' bit' if v.precision != -1 else 'native',
-                                                    str_dtype(v.dtype))
-                            if not isinstance(v.dtype, NativeBool)
-                            else  str_dtype(v.dtype) for v in types])
-            body.append(IfSection(LiteralTrue(), [PyErr_SetString('PyExc_TypeError', '"{} must be {}"'.format(var_name, error)), Return([LiteralInteger(0)])]))
+                assert elem[2] == 0
+                body.append(IfSection(PyccelNot(elem[1]), [PyErr_SetString('PyExc_TypeError', f'"{var_name} must be {error}"'), Return([PyccelUnarySub(LiteralInteger(1))])]))
+            else:
+                arg_type_check_list.sort(key= lambda x : x[2])
+                for elem in arg_type_check_list:
+                    var_name = elem[0].name
+                    body.append(IfSection(elem[1], [AugAssign(check_var, '+' ,elem[2])]))
+                body.append(IfSection(LiteralTrue(), [PyErr_SetString('PyExc_TypeError', f'"{var_name} must be {error}"'), Return([PyccelUnarySub(LiteralInteger(1))])]))
             check_func_body += [If(*body)]
 
         check_func_body = [Assign(check_var, LiteralInteger(0))] + check_func_body
         check_func_body.append(Return([check_var]))
         # Creating check function definition
-        check_func_name = self.scope.parent_scope.get_new_name('type_check')
+        check_func_name = self.scope.parent_scope.get_new_name(f'type_check_{func_name}')
```

Comment:
Interesting... The argument `func_name` was not used in this function!

Reply:
Yes! This also surprised me. But I think it makes the code clearer to name the generated type_check function something related to the function that it is handling